tools/test_utils/fullservice.py

```diff
@@ -0,0 +1,213 @@
+        r = connection.getresponse()
+
+        try:
+            raw_response = r.read()
+            response_data = json.loads(raw_response)
+            print(f'request returned {response_data}')
+        except ValueError:
+            raise ValueError('API returned invalid JSON:', raw_response)
+
+        # TODO requests might be flakey due to timing issues... waiting 2 seconds to bypass most of these issues
+        time.sleep(2)
+        if 'error' in response_data:
+            return response_data['error']
+        else:
+            return response_data['result']
+
+    def import_account(self, mnemonic) -> bool:
+        print(f'importing full service account {mnemonic}')
+        params = {
+            'mnemonic': mnemonic,
+            'key_derivation_version': '2',
+        }
+        r = self._request({
+            "method": "import_account",
+            "params": params
+        })
+
+        if 'error' in r:
+            # If we failed due to a unique constraint, it means the account already exists
+            return 'Diesel Error: UNIQUE constraint failed' in r['error']['data']['details']
+        return True
+
+    # retrieve accounts from mobilecoin/target/sample_data/keys
+    def get_test_accounts(self) -> Tuple[str, str]:
+        print(f'retrieving accounts for account_keys_0 and account_keys_1')
+        keyfile_0 = open(os.path.join(constants.KEY_DIR, 'account_keys_0.json'))
+        keyfile_1 = open(os.path.join(constants.KEY_DIR, 'account_keys_1.json'))
+        keydata_0 = json.load(keyfile_0)
+        keydata_1 = json.load(keyfile_1)
+
+        keyfile_0.close()
+        keyfile_1.close()
+
+        return (keydata_0['mnemonic'], keydata_1['mnemonic'])
+
+    # check if full service is synced within margin
+    def sync_status(self, eps=5) -> bool:
+        # ping network
+        try:
+            r = self._request({
+                "method": "get_network_status"
+            })
+        except ConnectionError as e:
+            print(e)
+            return False
+
+        # network offline
+        if int(r['network_status']['network_block_height']) == 0:
+            return False
+
+        # network online
+        network_block_height = int(r['network_status']['network_block_height'])
+        local_block_height = int(r['network_status']['local_block_height'])
+
+        # network is acceptably synced
+        return (network_block_height - local_block_height < eps)
+
+    # retrieve wallet status
+    def get_wallet_status(self):
+        r = self._request({
+            "method": "get_wallet_status"
+        })
+        return r['wallet_status']
+
+    # ensure at least two accounts are in the wallet. Some accounts are imported by default, but the number changes.
+    def setup_accounts(self):
+        account_ids, account_map = self.get_all_accounts()
+        if len(account_ids) >= 2:
+            self.account_ids = account_ids
+            self.account_map = account_map
```

Comment:
These would get overridden by the `get_all_accounts` call below (at the end of the method). That call should probably live under the `else:` clause.